Sources/libcggen/BCCGGenerator.swift

```diff
@@ -0,0 +1,330 @@
+      pointer.withMemoryRebound(to: UInt8.self, capacity: count) { pt in
+        UnsafeBufferPointer(start: pt, count: count)
+      }
+    }
+    return Array(bytePtr)
+  }
+}
+
+extension CGFloat: ByteCodable {
+  var byteCode: [UInt8] {
+    Float(self).bitPattern.byteCode
+  }
+}
+
+extension Bool: ByteCodable {
+  var byteCode: [UInt8] {
+    if self {
+      return [1]
+    }
+    return [0]
+  }
+}
+
+extension CGPoint: ByteCodable {
+  var byteCode: [UInt8] {
+    x.byteCode + y.byteCode
+  }
+}
+
+extension Array: ByteCodable where Element: ByteCodable {
+  var byteCode: [UInt8] {
+    UInt32(count).byteCode + map { element in
+      element.byteCode
+    }.joined()
+  }
+}
+
+extension CGRect: ByteCodable {
+  var byteCode: [UInt8] {
+    x.byteCode + y.byteCode + width.byteCode + height.byteCode
+  }
+}
+
+extension CGPathFillRule: ByteCodable {
+  var byteCode: [UInt8] {
+    [UInt8(rawValue)]
+  }
+}
+
+extension DashPattern: ByteCodable {
+  var byteCode: [UInt8] {
+    phase.byteCode + lengths.byteCode
+  }
+}
+
+extension CGPathDrawingMode: ByteCodable {
+  var byteCode: [UInt8] {
+    [UInt8(rawValue)]
+  }
+}
+
+extension CGAffineTransform: ByteCodable {
+  var byteCode: [UInt8] {
+    let abcd = a.byteCode + b.byteCode + c.byteCode + d.byteCode
+    return abcd + tx.byteCode + ty.byteCode
+  }
+}
+
+extension CGLineJoin: ByteCodable {
+  var byteCode: [UInt8] {
+    [UInt8(rawValue)]
+  }
+}
+
+extension CGLineCap: ByteCodable {
+  var byteCode: [UInt8] {
+    [UInt8(rawValue)]
+  }
+}
+
+extension CGColorRenderingIntent: ByteCodable {
+  var byteCode: [UInt8] {
+    [UInt8(rawValue)]
+  }
+}
+
+extension RGBACGColor: ByteCodable {
+  var byteCode: [UInt8] {
+    red.byteCode + green.byteCode + blue.byteCode + alpha.byteCode
+  }
+}
+
+extension ByteCodable where Self == [(CGFloat, RGBACGColor)] {
+  var byteCode: [UInt8] {
+    UInt32(self.count).byteCode + self.map { (v: CGFloat, v2: RGBACGColor) in
+      v.byteCode + v2.byteCode
+    }.joined()
+  }
+}
+
+extension Gradient: ByteCodable {
+  var byteCode: [UInt8] {
+    locationAndColors.byteCode
+  }
+}
+
+extension CGSize: ByteCodable {
+  var byteCode: [UInt8] {
+    width.byteCode + height.byteCode
+  }
+}
+
+extension Shadow: ByteCodable {
+  var byteCode: [UInt8] {
+    offset.byteCode + blur.byteCode + color.byteCode
+  }
+}
+
+extension CGBlendMode: ByteCodable {
+  var byteCode: [UInt8] {
+    [UInt8(rawValue)]
+  }
+}
+
+extension CGGradientDrawingOptions: ByteCodable {
+  var byteCode: [UInt8] {
+    [UInt8(rawValue)]
+  }
+}
+
+private func byteCommand(_ code: UInt8, _ args: ByteCodable...) -> [UInt8] {
+  [code] + args.map { arg -> [UInt8] in
+    arg.byteCode
+  }.joined()
+}
+
+private func generateSteps(steps: [DrawStep]) -> [UInt8] {
+  Array(steps.map{ step in step.byteCode}.joined())
+}
+
+extension DrawStep: ByteCodable {
+  var byteCode: [UInt8] {
+    switch self {
+    case .saveGState:
+      return byteCommand(0)
```

Comment:
Instead of being hardcoded, value should be taken from `BCCommon.Command` enum:
```
private func byteCommand(_ command: BCCommon.Command, _ args: ByteCodable...) -> [UInt8] {
   [command.rawValue] + ....
}
....
switch self {
     case .saveGState:
       return byteCommand(.saveGState)
     case .restoreGState:
       return byteCommand(.restoreGState)
     case let .moveTo(to):
       return byteCommand(.moveTo, to)
```